rct229/rules/section6/section6rule5.py

```diff
@@ -0,0 +1,112 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.schema.config import ureg
+from rct229.utils.masks import invert_mask
+
+BUILDING_AREA_CUTTOFF = ureg("5000 ft2")
+
+
+class Section6Rule5(RuleDefinitionListIndexedBase):
+    """Rule 5 of ASHRAE 90.1-2019 Appendix G Section 6 (Lighting)"""
+
+    def __init__(self):
+        super(Section6Rule5, self).__init__(
+            id="6-5",
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            each_rule=Section6Rule5.BuildingRule(),
+            index_rmr="baseline",
+            description="Baseline building is modeled with automatic shutoff controls in buildings >5000 sq.ft.",
+            required_fields={
+                "$": ["calendar", "schedules"],
+                "calendar": ["is_leap_year"],
+            },
+            list_path="ruleset_model_instances[0].buildings[*]",
+            data_items={
+                "is_leap_year_b": ("baseline", "calendar/is_leap_year"),
+                "schedules_b": ("baseline", "schedules"),
+                "schedules_p": ("proposed", "schedules"),
+            },
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section6Rule5.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                each_rule=Section6Rule5.BuildingRule.ZoneRule(),
+                index_rmr="baseline",
+                required_fields={"$": ["building_open_schedule"]},
+                data_items={
+                    "building_open_schedule_b",
+                    ("baseline", "building_open_schedule"),
+                },
+            )
+
+        def is_applicable(self, context, data):
+            building_total_area_b = pint_sum(
+                "$..spaces[*].floor_area", building_b, ZERO.AREA
+            )
+
+            return building_total_area_b > BUILDING_AREA_CUTTOFF
+
+        class ZoneRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section6Rule5.BuildingRule.ZoneRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    each_rule=Section6Rule5.BuildingRule.ZoneRule.SpaceRule(),
+                    index_rmr="baseline",
+                )
+
+            def create_data(self, context, data=None):
+                return {
+                    "avg_zone_height_b": get_avg_zone_height(zone_b),
+                    "avg_zone_height_p": get_avg_zone_height(zone_p),
```

Comment:
Please kindly ignore if I'm wrong, Charlie. 
Shouldn't ```zone_p``` and ```zone_b``` be defined before (e.g., zone_ b = context.baseline in line 61)?

Reply:
Yes, thank you.